storage/wdpost/wdpost_run_faults.go

```diff
@@ -92,6 +93,31 @@ func (s *WindowPoStScheduler) declareRecoveries(ctx context.Context, dlIdx uint6
 			continue
 		}
 
+		// rules to follow if we have indicated that we don't want to recover more than X sectors in a deadline
+		if recoveringSectorLimit > 0 {
+			// something weird happened, break because we can't recover any more
+			if recoveringSectorLimit < totalSectorsToRecover {
+				log.Warnf("accepted more recoveries (%d) than recoveringSectorLimit (%d)", totalSectorsToRecover, recoveringSectorLimit)
+				break
+			}
+
+			maxNewRecoverable := recoveringSectorLimit - totalSectorsToRecover
+
+			// we need to trim the recover bitfield
+			if recoveredCount > maxNewRecoverable {
+				recoverySlice, err := recovered.All(math.MaxUint64)
+				if err != nil {
+					log.Errorw("failed to slice recovery bitfield, breaking out of recovery loop", err)
+					break
+				}
+
+				log.Warnf("only adding %d sectors to respect recoveringSectorLimit %d", maxNewRecoverable, recoveringSectorLimit)
+
+				recovered = bitfield.NewFromSet(recoverySlice[0:maxNewRecoverable])
```

Comment:
Technically that `0` isn't needed
```suggestion
				recovered = bitfield.NewFromSet(recoverySlice[:maxNewRecoverable])
```